Sources/AsyncDNSResolver/dnssd/DNSResolver_dnssd.swift

```diff
@@ -138,32 +123,54 @@ struct DNSSD {
                 handler.handleRecord(errorCode: errorCode, data: rdata, length: rdlen)
             }
 
-            let serviceRefPtr = UnsafeMutablePointer<DNSServiceRef?>.allocate(capacity: 1)
-            defer { serviceRefPtr.deallocate() }
+            let serviceRefPointer = UnsafeMutablePointer<DNSServiceRef?>.allocate(capacity: 1)
+
+            // Wrap 'handler' into a pointer so we can pass it to DNSServiceQueryRecord
+            let replyHandlerPointer = UnsafeMutablePointer<QueryReplyHandler>.allocate(capacity: 1)
+            replyHandlerPointer.initialize(repeating: handler, count: 1)
```

Comment:
Let's avoid `initialize(repeating:)`, it's not needed. Just [`initialize(to:)`](https://developer.apple.com/documentation/swift/unsafemutablepointer/initialize(to:)) is fine.

Reply:
Fixed in latest commit